source/core/lighting/photons.cpp

```diff
@@ -2280,18 +2280,21 @@ void PhotonGatherer::gatherPhotonsRec(int start, int end)
 
     // check this photon
 
-    // find distance from pt
-    ptToPhoton = - (*pt_s) + Vector3d(photon->Loc);
-    // all distances are squared
-    dx = ptToPhoton[X]*ptToPhoton[X];
-    dy = ptToPhoton[Y]*ptToPhoton[Y];
-    dz = ptToPhoton[Z]*ptToPhoton[Z];
+    // find distance in DimToUse from pt
+    d = photon->Loc[DimToUse] - (*pt_s)[DimToUse];
+    d *= d;
 
-    if (!(  ((dx>dmax_s) && ((DimToUse)==X)) ||
-            ((dy>dmax_s) && ((DimToUse)==Y)) ||
-            ((dz>dmax_s) && ((DimToUse)==Z)) ))
+    if (d<dmax_s)
     {
-        // it fits manhatten distance - maybe we can use this photon
+        // it fits manhatten, DimToUse distance - maybe we can use this photon
+
+        dx = photon->Loc[X] - (*pt_s)[X];
+        dy = photon->Loc[Y] - (*pt_s)[Y];
+        dz = photon->Loc[Z] - (*pt_s)[Z];
+        ptToPhoton = Vector3d(dx,dy,dz);
+        dx *= dx;
+        dy *= dy;
+        dz *= dz;
```

Comment:
Since `dx`, `dy` and `dz` are not subsequently used (correct me if I'm mistaken), the above and the next statement could be collapsed to:

    ptToPhoton = Vector3d(photon->Loc) - *pt_s;
    d = ptToPhoton.lengthSqr();